app/src/main/java/com/duckduckgo/app/browser/WebNavigationState.kt

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2019 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser
+
+import android.webkit.WebBackForwardList
+import androidx.core.net.toUri
+import com.duckduckgo.app.global.isHttpsVersionOfUri
+
+
+interface WebNavigationState {
+    val originalUrl: String?
+    val currentUrl: String?
+    val title: String?
+    val stepsToPreviousPage: Int
+    val canGoBack: Boolean
+    val canGoForward: Boolean
+    val hasNavigationHistory: Boolean
+}
+
+sealed class WebNavigationStateChange
```

Comment:
Elsewhere, we tend to _block_ the subclasses of a sealed class using `{ }`. i think it makes it more readable. e.g,

    sealed class WebNavigationStateChange {
        data class NewPage(val url: String, val title: String?) : WebNavigationStateChange()
        data class UrlUpdated(val url: String) : WebNavigationStateChange()
        object PageCleared : WebNavigationStateChange()
        object Unchanged : WebNavigationStateChange()
        object Other : WebNavigationStateChange()
    }

Reply:
I think that looks nicer too 👍